### DISCUSSION TITLE
Issue with including dynamic menu

### DISCUSSION THREAD

**Discussion by mrosenov** [smarty-php/smarty#1091] [Q&A]
I'm having a issue with dynamic menu so the problem is that when the menu.tpl is populated with static info its shown correctly in my  layout.tpl, but when its populated with data from database is not shown at all like the assigned variable is empty. And this is only when I'm visiting any page unless I open url/menu.php then everything is fine. I will attach images at the bottom
menu.tpl
```
<nav>
    <ul>
        <li><a href="/test">{$info}</a></li>
        <li><a href="/test2">test2</a></li>
    </ul>
</nav>
```
menu.php
```
<?php
global $smarty;

$smarty->assign('info', new InfoPage(2) );
$smarty->display('menu.tpl');
```

layout.tpl
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>title</title>
    <link rel="icon" href="{$RESOURCE}images/favicon.ico" type="image/x-icon">
    <link rel="stylesheet" href="{$RESOURCE}css/style.css">
</head>
<body>
<header>
    <h1>Header Content</h1>
    {include file="menu.tpl"}
</header>
<main>
    {block "content"}{/block}
</main>
<footer>
    <h1>Footer Content</h1>
</footer>
</body>
</html>
```
This is when I'm visiting my index or any other page 
![image](https://github.com/user-attachments/assets/72272dc5-ff99-4141-a2cd-a89d09666ead)
And this is when I visit menu.php
![image](https://github.com/user-attachments/assets/68139c69-9e0c-4429-ac35-3b86989465be)

**Comment by wisskid** [comment]
If you don't assign anything to info, e.g. like you do with `$smarty->assign('info', new InfoPage(2) );` in menu.php, your menu entry will remain empty.

**Reply by mrosenov** [reply]
Of course I do assign. Anyway I figured out a work around using a helper object.
